src/replication.c

```diff
@@ -491,6 +491,70 @@ start_rebuild(void *buf, replica_t *replica, uint64_t data_len)
 	return ret;
 }
 
+/*
+ * Compare time..
+ * results: (T1 > T2) ? T1 : T2
+ */
+static int
+compare_time(struct timespec t1, struct timespec t2)
+{
+	if (t1.tv_sec > t2.tv_sec)
+		return 1;
+
+	if ((t1.tv_sec == t2.tv_sec) &&
+	    (t1.tv_nsec > t2.tv_nsec))
+		return 1;
+	return 0;
+}
+
+static int
+check_for_old_ios(spec_t *spec, replica_t *replica, struct timespec last)
+{
+	struct timespec diff1, diff2 = { 0 }, now;
+	int ret = 0;
+	replica_t *r;
+	rcmd_t *cmd;
+
+	ASSERT(MTX_LOCKED(&spec->rq_mtx));
+
+	TAILQ_FOREACH(r, &spec->rq, r_next) {
+		if ((replica && r == replica) || (!replica)) {
+			cmd = TAILQ_FIRST(&r->readyq);
```

Comment:
there are chances that this 'cmd' might have got deleted in replica_thread.

Reply:
yes.